scripts/rename_resources.py

```diff
@@ -26,23 +27,17 @@
 }
 
 
-def terraform_state(command: str, *args: str) -> bytes:
-    proc = subprocess.run(['terraform', 'state', command, *args],
-                          check=False,
-                          capture_output=True,
-                          shell=False)
-    sys.stderr.buffer.write(proc.stderr)
-    if proc.returncode == 0:
-        return proc.stdout
-    elif (
-        proc.returncode == 1
-        and command == 'list'
-        and b'No state file was found!' in proc.stderr
-    ):
-        log.info('No state file was found, assuming empty list of resources.')
-        return b''
+def terraform_state_list() -> list[str]:
+    try:
+        output = terraform.run('state', 'list')
+    except subprocess.CalledProcessError as e:
+        if e.returncode == 1 and 'No state file was found' in e.stderr:
+            log.info('No state file was found, assuming empty list of resources.')
+            return ['']
```

Comment:
This is not an empty list. On `develop` the return value is b`''` and `b''.decode().splitlines()` gives `[]`.

Reply:
> Please provide details on how you tested this.

Tested by adding a 
```
raise subprocess.CalledProcessError(1, '', '', 'No state file was found FOO')
```
inside the `try` block running the terraform state list. With this in place, when run the exception was caught and the `if` condition matched to return the "empty list of resources.".

Unfortunately, the `['']` return value then silently succeeded when iterated in `main()` since an empty string as `current_name` threw a KeyError, but it is ignored:
```
        for current_name in current_names:
            try:
                new_name = renamed[current_name]
            except KeyError:
                if current_name in renamed.values():
                    log.info('Found %r, already renamed', current_name)
            else:
                …
```